helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1742,18 +1742,32 @@ public boolean enableWagedRebalance(String clusterName, List<String> resourceNam
    */
   private boolean validateWeightForResourceConfig(ClusterConfig clusterConfig,
       ResourceConfig resourceConfig, IdealState idealState) {
-    // If ResourceConfig is null AND the default partition weight map is defined, we consider this resource valid
     if (resourceConfig == null) {
-      return !clusterConfig.getDefaultPartitionWeightMap().isEmpty();
-      // ResourceConfig is null and no default partition weight map has been defined. Not valid.
+      if (clusterConfig.getDefaultPartitionWeightMap().isEmpty()) {
+        logger.error(
+            "ResourceConfig for {} is null, and there are no default weights set in ClusterConfig!",
+            idealState.getResourceName());
+        return false;
+      }
+      // If ResourceConfig is null AND the default partition weight map is defined, and the map has all the required keys, we consider this valid since the default weights will be used
+      // Need to check the map contains all the required keys
+      if (clusterConfig.getDefaultPartitionWeightMap().keySet()
+          .containsAll(clusterConfig.getInstanceCapacityKeys())) {
+        // Contains all the required keys, so consider it valid since it will use the default weights
+        return true;
+      }
+      logger.error(
+          "ClusterConfig's default partition weight map doesn't have all the required keys!");
```

Comment:
nit, "ResourceConfig for {} is null, and ClusterConfig's default partition weight map doesn't have all the required keys!"

Reply:
Updated.